Packages/Network/Sources/Network/Client.swift

```diff
@@ -28,42 +36,52 @@ public final class Client: ObservableObject, Equatable, Identifiable, Hashable {
 
   public let server: String
   public let version: Version
-  public private(set) var connections: Set<String>
   private let urlSession: URLSession
   private let decoder = JSONDecoder()
 
-  /// Only used as a transitionary app while in the oauth flow.
-  private var oauthApp: InstanceApp?
-
-  private var oauthToken: OauthToken?
+  // Putting all mutable state inside an `OSAllocatedUnfairLock` makes `Client`
+  // provably `Sendable`. The lock is a struct, but it uses a `ManagedBuffer`
+  // reference type to hold its associated state.
+  private let critical: OSAllocatedUnfairLock<Critical>
```

Comment:
Ah this is nice. I was wondering how to make client mutable properties isolated. Do you think it's a good way to do this moving forward? Or should `Client` have a nested actor `ClientValues` or something like that?

Reply:
A nested `actor` type would work, though you'd have to do some dancing around it to make things like `id` remain synchronous. In this case I'd favour the lock, as it will perform the same job with a little less overhead and a lot less changes in the code & API.